frontend/src/entities/jobs/algorithms.tsx

```diff
@@ -71,83 +73,114 @@ export type Hyperparameter = {
     converter?: (input: any) => any;
 };
 
-const floatValidator = (required = false, positive = false) => {
-    let number_context = z.number({
-        invalid_type_error: `Must be a float${required ? ' and is required.' : ''}`,
+const floatValidator = (fieldName: string, required = false, positive = false) => {
+    return numberValidator(fieldName, {
+        isFloat: true,
+        required: required,
+        min: positive ? 0 : Number.NEGATIVE_INFINITY,
+        includeMin: false // If positive, can't be zero, will never reach Number.NEGATIVE_INFINITY regardless
     });
+};
 
-    if (positive) {
-        number_context = number_context.min(0, { message: 'Must be at least 0' });
-    }
-
-    let context: any = z.preprocess(Number, number_context);
-    if (!required) {
-        context = context.optional().or(z.literal(''));
-    }
-    return context;
+const positiveIntValidator = (fieldName: string, required = true, minimum = 1, inputProps : rangeValidatorProps = {}) => {
+    return numberValidator(fieldName, 
+        {
+            min: minimum,
+            required: required,
+            ...inputProps
+        })
 };
 
-const positiveIntMessage = (field: string, required = true) =>
-    `${field} ${required ? 'is required and ' : ''}must be a positive integer`;
+type rangeValidatorProps = {
+    min?: number;
+    max?: number;
+    isFloat?: boolean;
+    required?: boolean;
+    allowAuto?: boolean;
```

Comment:
Is there a reason for `allowAuto` and `alternateValues`? Can we just include `auto` in the alternate values array?

Reply:
No reason other than trying to stay consistent with previous. Consolidating into just alternateValues is probably preferred, so I will do that.